src/main/java/dao/BookRepositoryImpl.java

```diff
@@ -1,43 +1,53 @@
 package dao;
 
 import exception.DataProcessingException;
-import java.awt.print.Book;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.EntityTransaction;
 import java.util.List;
-import org.hibernate.Session;
-import org.hibernate.SessionFactory;
-import org.hibernate.Transaction;
+import java.util.Optional;
+import model.Book;
+import org.springframework.stereotype.Repository;
 
+@Repository
 public class BookRepositoryImpl implements BookRepository {
-    private SessionFactory sessionFactory;
+    private final EntityManagerFactory entityManagerFactory;
+
+    public BookRepositoryImpl(EntityManagerFactory entityManagerFactory) {
+        this.entityManagerFactory = entityManagerFactory;
+    }
 
     @Override
     public Book save(Book book) {
-        Session session = null;
-        Transaction transaction = null;
-        try {
-            session = sessionFactory.openSession();
-            transaction = session.beginTransaction();
-            session.persist(book);
+        EntityTransaction transaction = null;
+        try (EntityManager entityManager = entityManagerFactory.createEntityManager()) {
+            transaction = entityManager.getTransaction();
+            entityManager.persist(book);
             transaction.commit();
         } catch (Exception e) {
             if (transaction != null) {
                 transaction.rollback();
             }
-            throw new DataProcessingException("Can't save book to DB", e);
-        } finally {
-            if (session != null) {
-                session.close();
-            }
+            throw new DataProcessingException("Can't save book to DB");

```

Comment:
it makes sense to pass the root cause:
```suggestion
            throw new DataProcessingException("Can't save book to DB", e);

```